<launch>
  <arg name="num_vehicles" default="3" />
  <arg name="bind_ip" default="0.0.0.0" />

  <!-- Per-vehicle network args (override as needed). Up to 6 vehicles supported -->
  <arg name="vehicle_1_ip" default="10.15.129.5" /> <!--초록-->
  <arg name="vehicle_2_ip" default="10.15.129.4" /> <!-- 빨강 -->
  <arg name="vehicle_3_ip" default="10.15.129.51" /> <!-- 보라 -->
  <arg name="vehicle_4_ip" default="127.0.0.1" />
  <arg name="vehicle_5_ip" default="127.0.0.1" />
  <arg name="vehicle_6_ip" default="127.0.0.1" />
  <!-- Vehicle color hints (used by BEV teleporter via shared args) -->
  <arg name="vehicle_1_color" default="green" />
  <arg name="vehicle_2_color" default="pink" />
  <arg name="vehicle_3_color" default="purple" />
  <arg name="vehicle_4_color" default="" />
  <arg name="vehicle_5_color" default="" />
  <arg name="vehicle_6_color" default="" />

  <node pkg="carla_multi_vehicle_planner" type="udp_ackermann_sender.py" name="udp_ackermann_sender" output="screen">
    <param name="bind_ip" value="$(arg bind_ip)" />
    <param name="num_vehicles" value="$(arg num_vehicles)" />
    <param name="send_angle_as_float" value="true" />

    <!-- Per-vehicle topics (defaults align with CARLA bridge). Override if needed. -->
    <param name="vehicles/ego_vehicle_1/topic" value="/carla/ego_vehicle_1/vehicle_control_cmd" />
    <param name="vehicles/ego_vehicle_2/topic" value="/carla/ego_vehicle_2/vehicle_control_cmd" />
    <param name="vehicles/ego_vehicle_3/topic" value="/carla/ego_vehicle_3/vehicle_control_cmd" />
    <param name="vehicles/ego_vehicle_4/topic" value="/carla/ego_vehicle_4/vehicle_control_cmd" />
    <param name="vehicles/ego_vehicle_5/topic" value="/carla/ego_vehicle_5/vehicle_control_cmd" />
    <param name="vehicles/ego_vehicle_6/topic" value="/carla/ego_vehicle_6/vehicle_control_cmd" />

    <!-- Network destinations -->
    <param name="vehicles/ego_vehicle_1/dest_ip" value="$(arg vehicle_1_ip)" />
    <param name="vehicles/ego_vehicle_1/dest_port" value="5555" />
    <param name="vehicles/ego_vehicle_2/dest_ip" value="$(arg vehicle_2_ip)" />
    <param name="vehicles/ego_vehicle_2/dest_port" value="5555" />
    <param name="vehicles/ego_vehicle_3/dest_ip" value="$(arg vehicle_3_ip)" />
    <param name="vehicles/ego_vehicle_3/dest_port" value="5555" />
    <param name="vehicles/ego_vehicle_4/dest_ip" value="$(arg vehicle_4_ip)" />
    <param name="vehicles/ego_vehicle_4/dest_port" value="5555" />
    <param name="vehicles/ego_vehicle_5/dest_ip" value="$(arg vehicle_5_ip)" />
    <param name="vehicles/ego_vehicle_5/dest_port" value="5555" />
    <param name="vehicles/ego_vehicle_6/dest_ip" value="$(arg vehicle_6_ip)" />
    <param name="vehicles/ego_vehicle_6/dest_port" value="5555" />

    <!-- Per-vehicle calibration (defaults). Override as needed per car. -->
    <param name="vehicles/ego_vehicle_1/angle_scale" value="1.0" />
    <param name="vehicles/ego_vehicle_1/angle_clip" value="50" />
    <param name="vehicles/ego_vehicle_1/angle_min_abs" value="0" />
    <param name="vehicles/ego_vehicle_1/angle_invert" value="false" />
    <param name="vehicles/ego_vehicle_1/angle_center_rad" value="-0.00" />
    <param name="vehicles/ego_vehicle_1/speed_scale" value="0.93" />
    <param name="vehicles/ego_vehicle_1/speed_min_abs" value="1.0" />
    <param name="vehicles/ego_vehicle_1/force_min_speed_on_zero" value="true" />
    <param name="vehicles/ego_vehicle_1/zero_speed_value" value="0.0" />

    <param name="vehicles/ego_vehicle_2/angle_scale" value="1.0" />
    <param name="vehicles/ego_vehicle_2/angle_clip" value="50" />
    <param name="vehicles/ego_vehicle_2/angle_min_abs" value="0" />
    <param name="vehicles/ego_vehicle_2/angle_invert" value="false" />
    <param name="vehicles/ego_vehicle_2/angle_center_rad" value="-0.00" />
    <param name="vehicles/ego_vehicle_2/speed_scale" value="1.0" />
    <param name="vehicles/ego_vehicle_2/speed_min_abs" value="1.0" />
    <param name="vehicles/ego_vehicle_2/force_min_speed_on_zero" value="true" />
    <param name="vehicles/ego_vehicle_2/zero_speed_value" value="2.5" />

    <param name="vehicles/ego_vehicle_3/angle_scale" value="1.3" />
    <param name="vehicles/ego_vehicle_3/angle_clip" value="50" />
    <param name="vehicles/ego_vehicle_3/angle_min_abs" value="0" />
    <param name="vehicles/ego_vehicle_3/angle_invert" value="false" />
    <param name="vehicles/ego_vehicle_3/angle_center_rad" value="-0.05" />
    <param name="vehicles/ego_vehicle_3/speed_scale" value="1.0" />
    <param name="vehicles/ego_vehicle_3/speed_min_abs" value="1.0" />
    <param name="vehicles/ego_vehicle_3/force_min_speed_on_zero" value="true" />
    <param name="vehicles/ego_vehicle_3/zero_speed_value" value="2.5" />

    <param name="vehicles/ego_vehicle_4/angle_scale" value="500.0" />
    <param name="vehicles/ego_vehicle_4/angle_clip" value="50" />
    <param name="vehicles/ego_vehicle_4/angle_min_abs" value="0" />
    <param name="vehicles/ego_vehicle_4/angle_invert" value="false" />
    <param name="vehicles/ego_vehicle_4/angle_center_rad" value="0.0" />
    <param name="vehicles/ego_vehicle_4/speed_scale" value="1.0" />
    <param name="vehicles/ego_vehicle_4/speed_min_abs" value="1" />
    <param name="vehicles/ego_vehicle_4/force_min_speed_on_zero" value="true" />
    <param name="vehicles/ego_vehicle_4/zero_speed_value" value="1" />

    <param name="vehicles/ego_vehicle_5/angle_scale" value="500.0" />
    <param name="vehicles/ego_vehicle_5/angle_clip" value="50" />
    <param name="vehicles/ego_vehicle_5/angle_min_abs" value="0" />
    <param name="vehicles/ego_vehicle_5/angle_invert" value="false" />
    <param name="vehicles/ego_vehicle_5/angle_center_rad" value="0.0" />
    <param name="vehicles/ego_vehicle_5/speed_scale" value="1.0" />
    <param name="vehicles/ego_vehicle_5/speed_min_abs" value="1" />
    <param name="vehicles/ego_vehicle_5/force_min_speed_on_zero" value="true" />
    <param name="vehicles/ego_vehicle_5/zero_speed_value" value="1" />

    <param name="vehicles/ego_vehicle_6/angle_scale" value="500.0" />
    <param name="vehicles/ego_vehicle_6/angle_clip" value="50" />
    <param name="vehicles/ego_vehicle_6/angle_min_abs" value="0" />
    <param name="vehicles/ego_vehicle_6/angle_invert" value="false" />
    <param name="vehicles/ego_vehicle_6/angle_center_rad" value="0.0" />
    <param name="vehicles/ego_vehicle_6/speed_scale" value="1.0" />
    <param name="vehicles/ego_vehicle_6/speed_min_abs" value="1" />
    <param name="vehicles/ego_vehicle_6/force_min_speed_on_zero" value="true" />
    <param name="vehicles/ego_vehicle_6/zero_speed_value" value="1" />
    <!-- Expose color metadata on the same namespace for tooling/debug -->
    <param name="vehicles/ego_vehicle_1/color" value="$(arg vehicle_1_color)" />
    <param name="vehicles/ego_vehicle_2/color" value="$(arg vehicle_2_color)" />
    <param name="vehicles/ego_vehicle_3/color" value="$(arg vehicle_3_color)" />
    <param name="vehicles/ego_vehicle_4/color" value="$(arg vehicle_4_color)" />
    <param name="vehicles/ego_vehicle_5/color" value="$(arg vehicle_5_color)" />
    <param name="vehicles/ego_vehicle_6/color" value="$(arg vehicle_6_color)" />
  </node>

</launch>

